crypto-jni/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+
+impl<T> From<jni::errors::Error> for EncryptedTranfersResult<T> {
+    fn from(_: jni::errors::Error) -> Self {
+        EncryptedTranfersResult::Err(3)
+    }
+}
+
+const AMOUNT_DECRYPTION_ERROR: i32 = 4;
+const PAYLOAD_CREATION_ERROR: i32 = 5;
+
+impl<T: serde::Serialize> EncryptedTranfersResult<T> {
+    fn to_jstring(&self, env: &JNIEnv) -> jstring {
+        let json_str = to_string(self).unwrap();
+        let out = env.new_string(json_str).unwrap();
+        out.into_inner()
+    }
+}
+
+#[derive(Serialize, SerdeSerialize, SerdeDeserialize)]
+#[serde(bound(serialize = "C: Curve", deserialize = "C: Curve"))]
+#[serde(rename_all = "camelCase")]
+struct JniInput<C: Curve> {
+    global: GlobalContext<C>,
+    amount: Amount,
+    sender_secret_key: elgamal::SecretKey<C>,
+    input_encrypted_amount: IndexedEncryptedAmount<C>,
+}
+
+type EncryptedTranfersInput = JniInput<ArCurve>;
+
+#[no_mangle]
+#[allow(non_snake_case)]
+/// The JNI wrapper for the `create_sec_to_pub_transfer` method.
+/// The `input` parameter must be a properly initalized `java.lang.String` that
+/// is non-null. The input must be valid JSON according to specified format
+pub extern "system" fn Java_com_concordium_sdk_crypto_encryptedtransfers_EncryptedTransfers_createSecToPubTransfer(
+    env: JNIEnv,
+    _: JClass,
+    input: JString,
+) -> jstring {
+    let input: EncryptedTranfersInput = match env.get_string(input) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => match from_str(rust_str) {
+                Ok(input) => input,
+                Err(err) => return Result::from(err).to_jstring(&env),
+            },
+            Err(err) => return Result::from(err).to_jstring(&env),
+        },
+        Err(err) => return Result::from(err).to_jstring(&env),
+    };
+
+    let decrypted_amount = match decrypt_encrypted_amount(
+        input.input_encrypted_amount.encrypted_chunks.clone(),
+        input.sender_secret_key.clone(),
+    ) {
+        EncryptedTranfersResult::Ok(amount) => amount,
+        EncryptedTranfersResult::Err(err) => return Result::Err(err).to_jstring(&env),
+    };
+
+    let input_amount: AggregatedDecryptedAmount<ArCurve> = AggregatedDecryptedAmount {
+        agg_encrypted_amount: input.input_encrypted_amount.encrypted_chunks,
+        agg_index: encrypted_transfers::types::EncryptedAmountAggIndex {
+            index: input.input_encrypted_amount.index.index,
+        },
+        agg_amount: decrypted_amount,
+    };
+
+    let mut csprng = thread_rng();
+
+    let payload = encrypted_transfers::make_sec_to_pub_transfer_data(
+        &input.global,
+        &input.sender_secret_key,
+        &input_amount,
+        input.amount,
+        &mut csprng,
+    );
+
+    match payload {
+        Some(payload) => EncryptedTranfersResult::Ok(payload).to_jstring(&env),
+        None => Result::Err(PAYLOAD_CREATION_ERROR).to_jstring(&env),
+    }
+}
+
+static TABLE_BYTES: &[u8] = include_bytes!("table_bytes.bin");
```

Comment:
I'm not very keen on having this checked in like this. I don't think this should just magically be present. 

At least the babystepgiantstep table must be documented on how it was constructed.